---
title: "E-CRISP to GFF"
author: "Jan Winter"
date: "20 April 2017"
output: html_document
---

# AIM

Convert E-CRISP re-evaluation table (results.tab) into GFF format for ENSEMBL usage


# Open Data

Downloaded results tab from E-CRISP re-evaluation is fomratted like this:

Name | Target-chrome(s) | Start | End | Gene targets | Spec-Score | Anno-Score | Eff-Score | Matchstring | Sequence | Direction | CDS_score | exon_score | seed_GC | doench_score | xu_score | doench_30_mer
--- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- |
KLHL22_GTCACGCAGTGTGGGTGTGA | 11 | 118928357 | 118928380 | ENSG00000110375::UPK2::+ | 76.5217391304348 | 3.125 | 40.1520887751803 | nMXMMMMMMMMMMMMMMMMXMMM | GTGACGCAGTGTGGGTGTGCCAG | fw | 0 | 0 | 0.7 | 0.0513522499590136 | 0.2562521888 | GTTCGTGACGCAGTGTGGGTGTGCCAGCGC

```{r opendata}
library(tidyverse)

path <- "/Users/janwinter/Downloads"
file <- "KLHL22_results.tab"

# load data as TIBBLE (data frame) with column names
ecrisp_table <- read_tsv(file = file.path(path,file), col_names = TRUE)

# show data
head(ecrisp_table)

```

# Convert Data

GFF format needs to be formatted like this: 

The first line of a GFF3 file must be a comment that identifies the version, e.g.

`##gff-version 3`  
Fields must be tab-separated. Also, all but the final field in each feature line must contain a value; "empty" columns should be denoted with a '.'

1. seqid - name of the chromosome or scaffold; chromosome names can be given with or without the 'chr' prefix. Important note: the seq ID must be one used within Ensembl, i.e. a standard chromosome name or an Ensembl identifier such as a scaffold ID, without any additional content such as species or assembly. See the example GFF output below.
2. source - name of the program that generated this feature, or the data source (database or project name)
3. type - type of feature. Must be a term or accession from the SOFA sequence ontology
4. start - Start position of the feature, with sequence numbering starting at 1.
5. end - End position of the feature, with sequence numbering starting at 1.
6. score - A floating point value.
7. strand - defined as + (forward) or - (reverse).
8. phase - One of '0', '1' or '2'. '0' indicates that the first base of the feature is the first base of a codon, '1' that the second base is the first base of a codon, and so on..
9. attributes - A semicolon-separated list of tag-value pairs, providing additional information about each feature. Some of these tags are predefined, e.g. ID, Name, Alias, Parent - see the GFF documentation for more details.


__Mapping from original data is as follows:__  

1. seqid = Target-chrome(s)
2. source = "E-CRISP"
3. type = "CRISPRtarget"
4. start = Start
5. end = End
6. score = doench_score
7. strand = Direction -> __needs to be converted: fw = + and rc = -__
8. phase = .
9. attributes = paste of __Matchstring	Sequence	Direction	CDS_score	exon_score	seed_GC	doench_score	xu_score	doench_30_mer__


```{r convert data}

gff_table <- tibble(
  "seqid" = ecrisp_table$`Target-chrome(s)`,
  "source" = "E-CRISP",
  "type" = "CRISPRtarget",
  "start" = ecrisp_table$Start,
  "end" = ecrisp_table$End,
  "score" = ecrisp_table$doench_score,
  "strand" = mutate(ecrisp_table, Direction=replace(Direction, Direction=="fw", "+")) %>% mutate(Direction=replace(Direction, Direction=="rc", "-")) %>% dplyr::select(Direction) %>% collect %>% .[["Direction"]],
  "phase" = ".",
  "attributes" = mutate(ecrisp_table, attributes = paste(paste("ID",Name, sep="="), paste("Targets",`Gene targets`, sep="="), paste("Matchstring",Matchstring, sep="="), paste("Sequence",Sequence, sep="="), paste("Direction",Direction, sep="="), paste("CDS",CDS_score, sep="="), paste("Exon-Score",exon_score, sep="="), paste("GCContent",seed_GC, sep="="), paste("Doench-Score",doench_score, sep="="), paste("CU-Score",xu_score, sep="="), paste("Doench_30mer",doench_30_mer, sep="="), sep=";")) %>% collect %>% .[["attributes"]]
)


print(gff_table)

```


# Save Data

Now we need to save the table as tab-separated file and add the comment `##gff-version 3` in the first line.  
In the table, all column names need to be removed!


```{r save}

# replace file extension
file2 <- sub(pattern = "tab", replacement = "tsv", x = file)

# remove file if present
file.remove(file.path(path, file2))

# open new file ndn write comment to is
cat("##gff-version 3\n", file = file.path(path, file2))


# save as TSV without colnames
readr::write_tsv(x = gff_table, path = file.path(path, file2),col_names = FALSE, append = TRUE)


```


