---
title: "TidyDF_CGIM"
author: "Jan Winter"
date: "30 Mai 2017"
output:
  html_document:
    fig_width: 10
    fig.retina: 1

---

This script is used to retrieve the following information from an analysis and data collection generated by Benedikt Rauscher.  
It uses:
- CGIM Interaction Information (May 2017)
- Cancer Cell line expression data (May 2017)

# Cell Line Expression Data

The cancer cell expression data is retrieved from COSMIC and CCLE and offers several features:  

| Cell Line | Symbol | expression | source | tissue |
| --------- | ------ | ---------- | ------ | ----- |
| HCT116 | CTNNB1 | 2.4 | CCLE | Colon |


```{r loadccex, include=FALSE}
# use devtools to install superheat
#devtools::install_github("rlbarter/superheat")
library(tidyverse)
library(DT)
library(superheat)

ccex <- read_tsv(file = "/Volumes/CRISPR_Screen/Scripts/CRISPR Analysis/CGIM/cell_lines_expression.tsv")

```



## Expression for FRMD5

First we only check for the expression levels of FRMD5 alone and group it by the tissue.

### FRMD members

```{r FRMD-ccex}
data <- ccex %>% dplyr::filter(grepl("FRMD.+", symbol) & source =="COSMIC" & cellline !="TT")

# now we make it for heat map
data.heat <- data.frame("cellline" = unique(data$cellline),
                        stringsAsFactors = FALSE)

# now we add for each contained gene a new column representing the expression data
genes <- unique(data$symbol)
for(i in 1: length(genes))
{
  old.names <- colnames(data.heat)
  datasub <- data %>% dplyr::filter(symbol == genes[i]) %>% dplyr::select(cellline, expr)
  data.heat <- merge.data.frame(data.heat, as.data.frame(datasub), by="cellline")
  colnames(data.heat) <- c(old.names,genes[i])
}

rownames(data.heat) <- data.heat$cellline
data.heat$cellline <- NULL

png("FRMD5.png", height = 8000, width = 7000,res = 1200, pointsize = 6)
# heatmap ALL
superheatmap <- superheat(data.heat,
          # change the size of the labels
          # retain original order of rows/cols
          pretty.order.rows = TRUE,
          pretty.order.cols = FALSE,
          # order the rows by FRMD5
          order.rows = order(data.heat$FRMD5),
          #heat.na.col = "white",
          row.dendrogram = TRUE,
          #col.dendrogram = TRUE,
          # generate  column clusters
          #n.clusters.rows = 5,
          clustering.method = "hierarchical",
          left.label = "variable",
          left.label.text.size = 0.2,
          force.left.label = TRUE,
          bottom.label.size = 0.2,
          bottom.label.text.angle = 45,
          bottom.label.text.size = 3,
          column.title = "Genes",
          row.title = "Cell Lines"
          
          #smooth.heat = TRUE
          )
dev.off()
# # heatmap defined cluster
# superheatmap <- superheat(data.heat,
#           # change the size of the labels
#           left.label.size = 0.4,
#           bottom.label.size = 0.1,
#           # retain original order of rows/cols
#           pretty.order.rows = TRUE,
#           pretty.order.cols = FALSE,
#           # order the rows by FRMD5
#           #order.rows = order(data.heat$FRMD5),
#           heat.na.col = "white",
#           #row.dendrogram = TRUE,
#           #col.dendrogram = TRUE,
#           # generate  column clusters
#           n.clusters.rows = 7,
#           clustering.method = "hierarchical"
#           #left.label = 'variable'
#           #smooth.heat = TRUE
#           )
# 
# superheatmap$membership.rows
# 
# data.members <- tibble(cellline = names(superheatmap$membership.rows))
# data.tmp <- data.frame("cellline" = names(superheatmap$membership.rows), stringsAsFactors = FALSE)
# data.tmp$cluster <- superheatmap$membership.rows
# data.members <- dplyr::left_join(data.members, as.tibble(data.tmp), by = "cellline")
# data.members <- unique(data.members %>% dplyr::left_join( dplyr::select(data, c(cellline,tissue)), by = "cellline"))
# 
# # PLot cluster
# p1 <- ggplot(data=data.members, aes(x=cluster)) +
#   geom_bar(aes(fill=tissue)) +
#     #geom_jitter(shape=16, position=position_jitter(0.2)) +
#   labs(title = "Normalized mRNA Expression in cell lines grouped by tissue for FRMD5", x = "Tissue", y = "Normalized mRNA expression") +
#     ggplot2::theme_bw() +
#     ggplot2::geom_hline(yintercept = 0) +
#     theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom")
# 
# p1


```

```{r FRMD5-ccex}


data <- ccex %>% dplyr::filter(symbol == "FRMD5")


# plot all samples against all selected genes
p1 <- ggplot(data=data, aes(x=reorder(tissue, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in cell lines grouped by tissue for FRMD5", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="none")

p1

ggsave(filename = "FRMD5_expression_tissue.png",plot = p1, device = "png",dpi = 300,width = 10)

# Now we just pick large_intestine
data <- data %>% dplyr::filter(tissue == "large_intestins")
p2 <- ggplot(data=data, aes(x=reorder(cellline, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines for FRMD5", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="none")

p2
ggsave(filename = "FRMD5_expression_large_intestine.png",plot = p2, device = "png",dpi = 300,width = 10)



# make a heatmap to see which cluster together
# not possible for one gene only

# Now we plot certain cell lines only
data <- data %>% dplyr::filter(cellline %in% c("HCT116","SW480","DLD1","RKO","HT29","HCT115","NCIH1299"))

ggplot(data=data, aes(x=reorder(cellline, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines for FRMD5", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom")

ggsave(filename = "FRMD5_expression_large_intestine_specific.png", device = "png",dpi = 300,width = 10)

```

# Gene Expression for B4GALT6/7

## All B4GALT

```{r B4GALT-ccex}

data <- ccex %>% dplyr::filter(grepl("B4GALT.+", symbol) & cellline !="TT")

# now we make it for heat map
data.heat <- data.frame("cellline" = unique(data$cellline),
                        stringsAsFactors = FALSE)

# now we add for each contained gene a new column representing the expression data
genes <- unique(data$symbol)
for(i in 1: length(genes))
{
  old.names <- colnames(data.heat)
  datasub <- data %>% dplyr::filter(symbol == genes[i]) %>% dplyr::select(cellline, expr)
  data.heat <- merge.data.frame(data.heat, as.data.frame(datasub), by="cellline")
  colnames(data.heat) <- c(old.names,genes[i])
}

rownames(data.heat) <- data.heat$cellline
data.heat$cellline <- NULL

png("B4GALT_all.png", height = 8000, width = 7000,res = 1200, pointsize = 6)
# heatmap ALL
superheatmap <- superheat(data.heat,
          # change the size of the labels
          # retain original order of rows/cols
          pretty.order.rows = TRUE,
          pretty.order.cols = TRUE,
          # order the rows by FRMD5
          #order.rows = order(data.heat$FRMD5),
          #heat.na.col = "white",
          row.dendrogram = TRUE,
          col.dendrogram = TRUE,
          # generate  column clusters
          #n.clusters.rows = 5,
          clustering.method = "hierarchical",
          left.label = "variable",
          left.label.text.size = 0.2,
          force.left.label = TRUE,
          bottom.label.size = 0.2,
          bottom.label.text.angle = 0,
          bottom.label.text.size = 3,
          column.title = "Genes",
          row.title = "Cell Lines"
          
          #smooth.heat = TRUE
          )
dev.off()

print(superheatmap$plot)

```

```{r B4GALT6-ccex}

data <- ccex %>% dplyr::filter(symbol == "B4GALT6")


# plot all samples against all selected genes
p1 <- ggplot(data=data, aes(x=reorder(tissue, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in cell lines grouped by tissue for B4GALT6", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom")

p1

ggsave(filename = "B4GALT6_expression_tissue.png",plot = p1, device = "png",dpi = 300,width = 10)


# Now we just pick large_intestine
data <- data %>% dplyr::filter(tissue == "large_intestine")
p2 <- ggplot(data=data, aes(x=reorder(cellline, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines for B4GALT6", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="none")

p2
ggsave(filename = "B4GALT6_expression_large_intestine.png",plot = p2, device = "png",dpi = 300,width = 10)

# Now we plot certain cell lines only
data <- data %>% dplyr::filter(cellline %in% c("HCT116","SW480","DLD1","RKO","HT29","HCT115","NCIH1299"))

ggplot(data=data, aes(x=reorder(cellline, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines for B4GALT6", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom")

ggsave(filename = "B4GALT6_expression_large_intestine_specific.png", device = "png",dpi = 300,width = 10)


```


```{r B4GALT7-ccex}

data <- ccex %>% dplyr::filter(symbol == "B4GALT7")


# plot all samples against all selected genes
p1 <- ggplot(data=data, aes(x=reorder(tissue, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in cell lines grouped by tissue for B4GALT7", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="none")

p1

ggsave(filename = "B4GALT7_expression_tissue.png",plot = p1, device = "png",dpi = 300,width = 10)

# Now we just pick large_intestine
data <- data %>% dplyr::filter(tissue == "large_intestine")
p2 <- ggplot(data=data, aes(x=reorder(cellline, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines for B4GALT7", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="none")

p2
ggsave(filename = "B4GALT7_expression_large_intestine.png",plot = p1, device = "png",dpi = 300,width = 10)

# Now we plot certain cell lines only
data <- data %>% dplyr::filter(cellline %in% c("HCT116","SW480","DLD1","RKO","HT29","HCT115","NCIH1299"))

ggplot(data=data, aes(x=reorder(cellline, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines for B4GALT7", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom")

ggsave(filename = "B4GALT7_expression_large_intestine_specific.png", device = "png",dpi = 300,width = 10)



```


# Gene Expression for PAQR7

## All PAQR

```{r PAQR-ccex}

data <- ccex %>% dplyr::filter(grepl("PAQR.+", symbol) & cellline !="TT")

# now we make it for heat map
data.heat <- data.frame("cellline" = unique(data$cellline),
                        stringsAsFactors = FALSE)

# now we add for each contained gene a new column representing the expression data
genes <- unique(data$symbol)
for(i in 1: length(genes))
{
  old.names <- colnames(data.heat)
  datasub <- data %>% dplyr::filter(symbol == genes[i]) %>% dplyr::select(cellline, expr)
  data.heat <- merge.data.frame(data.heat, as.data.frame(datasub), by="cellline")
  colnames(data.heat) <- c(old.names,genes[i])
}

rownames(data.heat) <- data.heat$cellline
data.heat$cellline <- NULL

png("PAQR_all.png", height = 8000, width = 7000,res = 1200, pointsize = 6)
# heatmap ALL
superheatmap <- superheat(data.heat,
          # change the size of the labels
          # retain original order of rows/cols
          pretty.order.rows = TRUE,
          pretty.order.cols = TRUE,
          # order the rows by FRMD5
          #order.rows = order(data.heat$FRMD5),
          #heat.na.col = "white",
          row.dendrogram = TRUE,
          col.dendrogram = TRUE,
          # generate  column clusters
          #n.clusters.rows = 5,
          clustering.method = "hierarchical",
          left.label = "variable",
          left.label.text.size = 0.2,
          force.left.label = TRUE,
          bottom.label.size = 0.2,
          bottom.label.text.angle = 0,
          bottom.label.text.size = 3,
          column.title = "Genes",
          row.title = "Cell Lines"
          
          #smooth.heat = TRUE
          )
dev.off()

print(superheatmap$plot)

```

```{r PAQR7-ccex}

data <- ccex %>% dplyr::filter(symbol == "PAQR7")


# plot all samples against all selected genes
p1 <- ggplot(data=data, aes(x=reorder(tissue, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in cell lines grouped by tissue for B4GALT6", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom")

p1

ggsave(filename = "PAQR7_expression_tissue.png",plot = p1, device = "png",dpi = 300,width = 10)


# Now we just pick large_intestine
data <- data %>% dplyr::filter(tissue == "large_intestine")
p2 <- ggplot(data=data, aes(x=reorder(cellline, expr), y=expr))+
  geom_boxplot() +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines for B4GALT6", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="none")

p2
ggsave(filename = "PAQR7_expression_large_intestine.png",plot = p2, device = "png",dpi = 300,width = 10)



```

```{r all}
data <- ccex %>% dplyr::filter(grepl("B4GALT6|B4GALT7|B4GALT1|PAQR7|FRMD5|CTNNB1", symbol) & cellline !="TT")

# Now we plot certain cell lines only
data <- data %>% dplyr::filter(cellline %in% c("HCT116","SW480","DLD1","RKO","HT29","HCT115","NCIH1299"))

ggplot(data=data, aes(x=cellline, y=expr))+
  geom_boxplot(aes(fill=symbol)) +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom") +
  facet_grid(symbol ~ .)

ggsave(filename = "ALL_expression_large_intestine_specific.png", device = "png",dpi = 300,width = 10, height = 10)



# check for CTNNB1, TCFexpression and canidates
data <- ccex %>% dplyr::filter(grepl("TMEM191B|KIF21A|KLHL22|RIC1|LIMS1|CEP55|CTNNB1|TCF7L1|TCF7L2", symbol) & cellline !="TT")

# Now we plot certain cell lines only
data <- data %>% dplyr::filter(cellline %in% c("HCT116","SW480","DLD1","RKO","HT29","HCT115","NCIH1299"))

ggplot(data=data, aes(x=cellline, y=expr))+
  geom_boxplot(aes(fill=symbol)) +
    geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Normalized mRNA Expression in colon cell lines", x = "Tissue", y = "Normalized mRNA expression") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom") +
  facet_grid(symbol ~ .)

ggsave(filename = "ALL_expression_large_intestine_OLDcandidates_relation.png", device = "png",dpi = 300,width = 10, height = 10)



```


# Gene-Gene Interaction Profiles

NEITHER FRMD5 NOR B4GALT6 had any significant outcomes!

These gene-gene interaction profiles are based on the CGIM analysis of Benedikt Rauscher and measure genetic interaction based on the essentiality (viability) of pooled CRISPR screenin data present in the GenomeCRISPR database.  

__CGIM1 contains the following information:__  

| Target Gene | Query Gene | AVG Fitness Wildtype | AVG Fitness Mutated | Log2FC (WT vs MUT) | P.value | FDR |
| ----------- | ---------- | -------------------- | ------------------- | ------------------ | ------- | --- |
TFAP2C	AFDN	6.911086054759725	5.133148892421228	-0.42906833522260657	2.4424906541753444e-15	5.986490858589377e-9


__CGIM2 contains the following information:__   

| Target Gene | Query Gene | Target Main Effect | Query Main Effect | Measured Fitnes (scaled) | Pi-Score |
| ----------- | ---------- | ------------------ | ----------------- | ------------------------ | ------- |



```{r loadcgim, include=FALSE}
#cgim1 <- read_tsv(file = "/Volumes/CRISPR_Screen/Scripts/CRISPR Analysis/CGIM/CGIM_Supplementary_Table_4_FULL.tsv")
#cgim2 <- read_tsv(file = "/Volumes/CRISPR_Screen/Scripts/CRISPR Analysis/CGIM/CGIM_Supplementary_Table_5_FULL.tsv")
```

# Check for Fitness Interactions

```{r fitness-WT-MUT}

#data <- cgim2 %>% dplyr::filter(`Target gene` == "PAQR7")



```


# Comisc MUTATION Data

```{r clpmut}
clpmut <- read_tsv(file = "/Users/janwinter/OneDrive/PhD/Followup/COSMIC/CosmicCLP_MutantExport.tsv")
clpmut$`Sample name` <- sub(pattern = "-|_",replacement = "",x = clpmut$`Sample name`)

#data <- clpmut %>% dplyr::filter(grepl("large_intestine", `Primary site`) & grepl("colon", `Site subtype 1`) &  `HGNC ID` %in% c("11998","6407","2514","583"))
#data <- clpmut %>% dplyr::filter(grepl("large_intestine", `Primary site`) &  `HGNC ID` %in% c("11998","6407","2514","583"))
data <- clpmut %>% dplyr::filter(`Sample name` %in% c("HCT116","SW48","DLD1","RKO","HT29","HCT115","NCIH1299") &  `HGNC ID` %in% c("11998","6407","2514","583"))

data


ggplot(data=data, aes(`Sample name`))+
  geom_bar(aes(`Sample name`,fill=`Gene name`)) +
    #geom_jitter(shape=16, position=position_jitter(0.2), aes(colour=source)) +
  labs(title = "Mutations in Colon Cell Lines", x = "Cell Lines", y = "Number of Mutations") +
    ggplot2::theme_bw() +
    ggplot2::geom_hline(yintercept = 0) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1), plot.title = element_text(size = rel(1.1)), legend.position="bottom") +
  facet_grid(`Gene name` ~ .)

ggsave(filename = "Colon_cells_mutations1.png", device = "png",dpi = 300,width = 10, height = 10)

```


```{r FRMD5}

# check for FRMD5 in `Gene name`
#data <- clpmut %>% dplyr::filter(`Gene name` == "B4GALT7")



```


